default.nix

```diff
@@ -358,152 +269,138 @@ let overrideCabal = pkg: f: if pkg == null then null else haskellLib.overrideCab
           sha256 = "13i6lz99x0jb9fgns7brlqnv5s5w4clp26l8c3kxd318r1krvr6w";
         }) {};
 
-        superconstraints =
-          # Remove override when assertion fails
-          assert (super.superconstraints or null) == null;
-          self.callPackage (self.haskellSrc2nix {
-            name = "superconstraints";
-            src = fetchurl {
-              url = "https://hackage.haskell.org/package/superconstraints-0.0.1/superconstraints.cabal";
-              sha256 = "0bgc8ldml3533522gp1x2bjiazllknslpl2rvdkd1k1zfdbh3g9m";
-            };
-            sha256 = "1gx9p9i5jli91dnvvrc30j04h1v2m3d71i8sxli6qrhplq5y63dk";
-          }) {};
-      } // (if enableLibraryProfiling then {
-        mkDerivation = expr: super.mkDerivation (expr // { enableLibraryProfiling = true; });
-      } else {});
+        mkDerivation = expr: super.mkDerivation (expr // {
+          inherit enableLibraryProfiling;
+        });
+      };
     };
     haskellOverlays = import ./haskell-overlays {
       inherit
         haskellLib
         nixpkgs jdk fetchFromGitHub
-        useReflexOptimizer stage2Script;
+        useReflexOptimizer
+        hackGet;
+      inherit (nixpkgs) lib;
       androidActivity = hackGet ./android-activity;
     };
-    stage2Script = nixpkgs.runCommand "stage2.nix" {
-      GEN_STAGE2 = builtins.readFile (nixpkgs.path + "/pkgs/development/compilers/ghcjs/gen-stage2.rb");
-      buildCommand = ''
-        echo "$GEN_STAGE2" > gen-stage2.rb && chmod +x gen-stage2.rb
-        patchShebangs .
-        ./gen-stage2.rb "${sources.ghcjs-boot}" >"$out"
-      '';
-      nativeBuildInputs = with nixpkgs; [
-        ruby cabal2nix
-      ];
-    } "";
-    ghcjsCompiler = ghc.callPackage (nixpkgs.path + "/pkgs/development/compilers/ghcjs/base.nix") {
-      bootPkgs = ghc;
-      ghcjsSrc = sources.ghcjs;
-      ghcjsBootSrc = sources.ghcjs-boot;
-      shims = sources.shims;
-      stage2 = import stage2Script;
+    ghcjs8_2_2Packages = nixpkgs.callPackage (nixpkgs.path + "/pkgs/development/haskell-modules") {
+      ghc = ghc8_2_2.ghcjs;
+      buildHaskellPackages = ghc8_2_2.ghcjs.bootPkgs;
+      compilerConfig = nixpkgs.callPackage (nixpkgs.path + "/pkgs/development/haskell-modules/configuration-ghc-7.10.x.nix") { inherit haskellLib; };
+      packageSetConfig = nixpkgs.callPackage (nixpkgs.path + "/pkgs/development/haskell-modules/configuration-ghcjs.nix") { inherit haskellLib; };
+      inherit haskellLib;
     };
-    ghcjsPackages = nixpkgs.callPackage (nixpkgs.path + "/pkgs/development/haskell-modules") {
-      ghc = ghcjsCompiler;
+    ghcjs8_4_3Packages = nixpkgs.callPackage (nixpkgs.path + "/pkgs/development/haskell-modules") {
```

Comment:
The GHCJS names here are misleading. GHCJS is not tied to 8.4.3 specifically, just the 8.4 branch. So at most the name should just refer to 8.4, potentially with a suffix to match `ghcjs.version`.

Reply:
Yes eventually I want to get Nixpkgs to just have an alias for ghc8_2, ghc8_4, etc. That way we don't have to update reflex-platform for each patch update to GHC.

Reply:
No, that's not what I mean. GHCJS does not have an 8.4.3 version. There's no relation between the third version component in GHCJS and GHC whatsoever.

Reply:
It should not be relevant. GHCJS uses its own submodule of GHC, NOT the `ghc` lib from the GHC you build it with. The only way the native GHC could affect things is if there are bugs that affect the behavior of programs it produces.

Reply:
Theoretically it shouldn't take much effort to make it possible to compile GHCJS 8.4 with GHC 8.2.

Reply:
Ok I just pushed a commit that drops the _patch version everywhere. This makes more sense to me & is more consistent with how the overlays work. I still want to get Nixpkgs to drop the patch release number since it doesn't seem too useful.